### PR TITLE
Archive unpopular / unsupported destinations

### PR DESCRIPTION
**Pull request by evantahler**
More information in the [doc](https://docs.google.com/document/d/1OL6vVaje5S1ajhXeSaGW3WKnvLik6g0ZDXnaNVDtAJs/edit).  Depends on https://github.com/airbytehq/airbyte/pull/35355 and https://github.com/airbytehq/airbyte-platform-internal/pull/11405.  Don't merge this until both of these are merged in.

This PR Should not be merged until all comments in the linked doc are resolved.

### CODE REVIEW

**Review by erohmensing — CHANGES_REQUESTED** [review]
Blocking until we are sure that `archived` will not break the platform ([comment](https://airbytehq-team.slack.com/archives/C02TL38U5L7/p1708711028486249?thread_ts=1708624845.955359&cid=C02TL38U5L7)) if an archived connector makes its way into a registry for any reason.

File: airbyte-integrations/connectors/destination-cassandra/metadata.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ data:
   name: Cassandra
   registries:
     cloud:
-      enabled: false # hide Cassandra Destination https://github.com/airbytehq/airbyte-cloud/issues/2606
+      enabled: true
```
[the remark this answers is not in the document]
Comment (evantahler) [reply airbyte-integrations/connectors/destination-cassandra/metadata.yaml]: ... great catch! it should not have been. 712df7e824094f4b9719ba54ce29950044638dc4

[the remark this answers is not in the document]
Comment (erohmensing) [reply airbyte-integrations/connectors/destination-cassandra/metadata.yaml]: Thanks! Easy mistake to make - that's why I'm being so pushy on making sure that `archived` in the registry doesn't break platform, even though it hypothetically should not be a problem if we never publish an archived connector to a registry.

**Comment by evantahler** [comment]
https://github.com/airbytehq/airbyte-platform-internal/pull/11405 is merged, so this should be ready to go!

File: airbyte-integrations/connectors/destination-cassandra/metadata.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ data:
     cloud:
       enabled: false # hide Cassandra Destination https://github.com/airbytehq/airbyte-cloud/issues/2606
```
Comment (erohmensing) [inline airbyte-integrations/connectors/destination-cassandra/metadata.yaml:13]:
```suggestion
      enabled: false
```

**Review by erohmensing — APPROVED** [review]
Approving as PRs resolving the potential breaking of the platform are merged. 

Thanks for creating a PR to handle hiding the archived connectors from the UI if they make it into a catalog. Please go ahead and remove them from the LaunchDarkly flag after that is merged! 

There is some stuff we could do if we _really_ want to do this right, in that we probably would want to make cloud and oss both aware that all these connectors are archived, so that if a user is using a connector that is now archived, they can see that in their UI. That would however involve releasing all the connectors (potentially new versions since they last hit cloud), cutting an OSS release so that OSS users have that info too. That's probably not ideal, and what we talked about re: one large catalog with all the info about which registries a connector is available in and if a connector is archived, would probably better solve this problem. I don't expect you to do this, but wanted to raise it in case we are wondering why we don't see these connectors as archived in the UI.

**Comment by octavia-approvington** [comment]
Approve this <br/> imagine the commander saying yes

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**